teachertool/src/components/AddressBar.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from "react";
+// eslint-disable-next-line import/no-internal-modules
+import css from "./styling/AddressBar.module.scss";
+
+import { useContext, useState, useMemo, useCallback, useEffect } from "react";
+import { AppStateContext } from "../state/appStateContext";
+import { classList } from "react-common/components/util";
+import { Input } from "react-common/components/controls/Input";
+import { loadProjectMetadataAsync } from "../transforms/loadProjectMetadataAsync";
+
+interface IProps {
+    className?: string;
+}
+
+export const AddressBar: React.FC<IProps> = ({ className }) => {
+    const { state: teacherTool } = useContext(AppStateContext);
```

Comment:
I believe this is a way of extracting `projectMetadata` _from_ the `teacherTool` object.

Reply:
I don't think I'm following. Naming this projectMetadata would mean accessing the actual project metadata like `projectMetadata.projectMetadata`.
Line 17 is where I extract `projectMetadata` to a local var.

Reply:
Oh, yeah, I was confused because the `projectMetadata` variable was accessed similar to how this is, and I thought we could just rename the state, but I realize now that it is a similar accessor, but we need both of these. I know I did not explain that well, but long story short, I understand now and appreciate the clarification!